raiden-ts/src/channels/epics.ts

```diff
@@ -798,71 +815,72 @@ export const channelDepositEpic = (
   }: RaidenEpicDeps,
 ): Observable<channelDeposit.failure> =>
   action$.pipe(
-    filter(isActionOf([channelDeposit.request, channelOpen.request])),
-    // ignore channelOpen.request if no deposit is requested on it
-    filter((action) => !channelOpen.request.is(action) || !!action.payload.deposit?.gt?.(0)),
-    groupBy((action) => action.meta.tokenNetwork), // serialize per tokenNetwork
+    filter(isActionOf(channelDeposit.request)),
+    groupBy((action) => action.meta.tokenNetwork),
     mergeMap((grouped$) =>
       grouped$.pipe(
-        mergeMap((action) => {
-          const error = new RaidenError(ErrorCodes.CNL_NO_OPEN_CHANNEL_FOUND, action.meta);
-          const notOpenAction = channelDeposit.failure(error, action.meta);
-          // this 'merge' re-emits the requests as soon as the channel is detected as 'open';
-          // emits deposit failure if channel isn't open or open request fails
-          return merge(
-            latest$.pipe(
-              pluck('state', 'channels', channelKey(action.meta)),
-              // if it's a open request, wait confirmed channel, otherwise pass to validate now
-              filter((channel) => (channelOpen.request.is(action) ? isntNil(channel) : true)),
-              map((channel) => {
-                if (channel?.state !== ChannelState.open) return notOpenAction;
-                return action; // re-emit deposit/open action when channel is open
-              }),
-            ),
-            action$.pipe(
-              filter(channelOpen.failure.is),
-              filter(
-                (failure) =>
-                  failure.meta.tokenNetwork === action.meta.tokenNetwork &&
-                  failure.meta.partner === action.meta.partner,
-              ),
-              mapTo(notOpenAction),
-            ),
-          ).pipe(take(1)); // for each request, take only first re-emited request or failure
-        }),
-        // concatMap to deposit on channels serially per tokenNetwork, in the order they're
-        // detected as 'open' and confirmed
+        // groupBy + concatMap ensure actions handling is serialized in a given tokenNetwork
         concatMap((action) =>
           combineLatest([latest$, config$]).pipe(
             first(),
             mergeMap(([{ state }, { subkey: configSubkey }]) => {
-              // if channelDeposit.failure, it means channel was not 'open', passthrough failure
-              if (channelDeposit.failure.is(action)) return of(action);
-
               const { tokenNetwork, partner } = action.meta;
-              // channel is guaranteed to be present and 'open', by mergeMap above
+
+              const token = findKey(state.tokens, (tn) => tn === tokenNetwork)! as Address;
```

Comment:
What is our guarantee that the token is already monitored? :thinking: Do we rely on the caller to check that first?

Reply:
`Raiden.openChannel` & `Raiden.depositChannel` do that, it's guaranteed, but it's also checked and will fail below if not, so it's safe